assets/src/js/IdleQueue.js

```diff
@@ -0,0 +1,73 @@
+// https://github.com/GoogleChromeLabs/idlize/blob/master/docs/IdleQueue.md
+import {IdleQueue} from 'idlize/IdleQueue.mjs';
+
+const XwpIdleQueue = {
+
+	/**
+	 * Lazy script selector.
+	 *
+	 * @type {string}
+	 */
+	queueSelector: '[data-queue-src]',
+
+	/**
+	 * Collection of src.
+	 *
+	 * @type {array}
+	 */
+	listSrc: [],
+
+	/**
+	 * Get src of script to queue.
+	 * Write these source in a function to write script in the head.
+	 * Push these functions in the queue.
+	 * Launch the queue
+	 *
+	 * @param {string} src The lazy sript src.
+	 * @returns {void}
+	 */
+	pushLinkScripts() {
+		const src = document.querySelectorAll( this.queueSelector );
+
+		src.forEach( ( s ) => {
+			this.listSrc.push( s.dataset.queueSrc );
+		} );
+
+		this.listSrc.forEach( ( source ) => {
+			const fun = function() {
+				let s = document.createElement( 'script' ),
+					el = document.getElementsByTagName( 'script' )[0];
+				s.async = 'true';
+				s.src = source;
+				el.parentNode.insertBefore( s, el );
+			};
+			window.xwpQueue.push( fun );
+		} );
+
+		this.pushInlineScripts();
+	},
+
+	/**
+	 * Send all functions in the queue.
+	 *
+	 * @returns {void}
+	 */
+	pushInlineScripts() {
+		const queue = new IdleQueue();
+		if ( window.xwpQueue ) {
+			window.xwpQueue.forEach( task =>
+				queue.pushTask( task, { time: 1000 } )
```

Comment:
I believe the time parameter should be epoch time, so more like `Date.now()`
https://github.com/GoogleChromeLabs/idlize/blob/master/docs/IdleQueue.md

Also wondering if this would cover this use case:
- putting into queue <script src='jquery.js'>
- putting next into queue <script>$('whatever jquery related')</script>

The concern that I have is that task gets evaluated (inserting jquery script to dom) but jquery is not loaded yet (scripts injected after document.close() are treated as async I believe, you also explicitly set async to true). 

If that's the case queue could be potentially evaluated with awaiting for onload event triggered by injected script like so (taken from another place, but should give you context)
```
await new Promise((resolve) => {
      const newScript = document.createElement('script')
      Array.from(script.attributes).forEach(({name, value}) => {
        newScript.setAttribute(name, value)
      })
      newScript.type = 'text/javascript'
      newScript.onload = () => {
        resolve()
      }
      script.parentNode.insertBefore(newScript, script)
    })
```

Also wondering if putting them all before first script wouldn't mess the order (reverse it) if they relate on each other (like having jquery plugin defined before jquery). 

Things to test I guess, they're more questions than actual issues (I haven't tested it)

Reply:
@MakiBM It's a valid concern. I guess it's worth testing out such a case.

In the `idlize` doc page you referenced, there's that code example:

```js
import {IdleQueue} from 'idlize/IdleQueue.mjs';

const queue = new IdleQueue();

queue.pushTask(() => {
  // Some expensive function that can run idly...
});

queue.pushTask(() => {
  // Some other task that depends on the above
  // expensive function having already run...
});
```

From that, it seems that the tasks are preserving order and are ran sequentially so this *should* be safe.

Reply:
The time parameter was coming from Rebel Mouse implementation inside https://www.brit.co/ but is not needed.
So I remove it.

And yes, maybe the page can be updated with jquery library and a plugin to test that case.